src/app/parcels/ActionBar/Statuses.tsx

```diff
@@ -83,7 +83,13 @@ const Statuses: React.FC<Props> = ({
     const [selectedStatus, setSelectedStatus] = useState<statusType | null>(null);
     const [statusModal, setStatusModal] = useState(false);
 
-    const selectedData = Array.from(selected.map((index) => data[index]));
+    const [selectedData, setSelectedData] = useState(
+        Array.from(selected.map((index) => data[index]))
+    );
```

Comment:
What's the reason behind having this as a state and updating it with useEffect? `selected` and `data` (we should've had a better name but alas) are both props so I didn't think this would be necessary.

Reply:
Ok I've been unable to recreate the error that caused me to implement this, so I think I will change this back to how it was before, but have described the error below for the record:

since adding the status table in the database as a subscription to the parcels table in the website, on deleting parcels, the data in the parcel table changes. Therefore selectedData, because it is defined in terms of data, changes too. This was causing the below error (which luckily was in my google search history)

`Cannot update a component (`HotReload`) while rendering a different component (`StatusesBarModal`). To locate the bad setState() call inside `StatusesBarModal, follow the stack trace as described]`

And I followed the solution in this Stack Exchange post https://stackoverflow.com/questions/62336340/cannot-update-a-component-while-rendering-a-different-component-warning

Not sure why the error is no longer showing... slightly concerned but at least we have the commit history to look back on if we need ?

Reply:
Think this error is unlikely to happen now as this modal no longer deals with this state